src/hammer-vlsi/technology/asap7/__init__.py

```diff
@@ -0,0 +1,156 @@
+                },
+            "SL": {
+                "lib": gdspy.GdsLibrary(),
+                "mvt_layer": 97
+                },
+            "SRAM": {
+                "lib": gdspy.GdsLibrary(),
+                "mvt_layer": 110
+                },
+        }
+        # create new libs
+        for vt, multi_lib in multi_libs.items():
+            multi_lib["lib"].name = asap7_original_gds.name.replace('SL', vt)
+
+        for cell in original_cells.values():
+            poly_dict = cell.get_polygons(by_spec=True)
+            # extract polygon from layer 100(the boundary for cell)
+            boundary_polygon = poly_dict[(100, 0)]
+            for vt, multi_lib in multi_libs.items():
+                mvt_layer = multi_lib['mvt_layer']
+                if mvt_layer:
+                    # copy boundary_polygon to mvt_layer to mark the this cell is a mvt cell.
+                    mvt_polygon = gdspy.PolygonSet(boundary_polygon, multi_lib['mvt_layer'], 0)
+                    mvt_cell = cell.copy(name=cell.name.replace('SL', vt), exclude_from_current=True, deep_copy=True).add(mvt_polygon)
+                else:
+                    # RVT, just copy the cell
+                    mvt_cell = cell.copy(name=cell.name.replace('SL', vt), exclude_from_current=True, deep_copy=True)
+                # add mvt_cell to corresponding multi_lib
+                multi_lib['lib'].add(mvt_cell)
+
+        for vt, multi_lib in multi_libs.items():
+            # write multi_lib with 2.5e-10 precision
+            multi_lib['lib'].precision = 2.5e-10
+            multi_lib['lib'].write_gds(os.path.splitext(orig_gds)[0] + '_' + vt + '.gds')
+
+    def fix_sram_cdl_bug(self) -> None:
+        """
+        vendor's SRAM cdl use slvt cell, this patch will sed cells name in which, fix this bug.
+        """
+        self.logger.info("sed slvt to sram in asap7_75t_SRAM.cdl")
+        pattern0 = re.compile("SL")
+        pattern1 = re.compile("slvt")
+
+        with tempfile.NamedTemporaryFile(delete=False) as tf:
+            sram_cdl = os.path.join(self.extracted_tarballs_dir, "ASAP7_PDKandLIB.tar/ASAP7_PDKandLIB_v1p5/asap7libs_24.tar.bz2/asap7libs_24/cdl/lvs/asap7_75t_SRAM.cdl")
+            with open(sram_cdl, 'r') as f:
+                tf.write(pattern1.sub("sram", pattern0.sub("SRAM", f.read())).encode('utf-8'))
+            shutil.copystat(sram_cdl, tf.name)
+            shutil.copy(tf.name, sram_cdl)
+
+    def post_par_script(self, d: dict) -> None:
+        """
+        Scale the final GDS by a factor of 4
+        Called by the `run_par` method
+        """
+        self.logger.info("Scaling down place & routed GDS")
+
+        # load original_gds
+        par_gds_file = d.get("par.outputs.output_gds")
+        gds_lib = gdspy.GdsLibrary(infile=par_gds_file)
+        # Iterate through cells that aren't part of standard cell library and scale
+        for k,v in gds_lib.cell_dict.items():
+            if not 'ASAP7_75t' in k:
```

Comment:
Is `startswith` still a correct filter? I'm worried people might develop other cells that are meant to work with the  `ASAP7_75t` cells and therefore have that string in their cell names. If we can narrow the filter that would help a bit.

Reply:
I ended up grabbing the cell list from the PDK's GDS folder to check against specific cells. This way people naming cells with `ASAP7_75t` in them won't be excluded.